### DISCUSSION TITLE
【map】how to change map service of immich

### DISCUSSION THREAD

**Discussion by Alberteins** [immich-app/immich#14511] [Q&A]
how to change map service of immich? the default map service works not very well in my country, so I want to change it to some local map service provider. it will be very nice of you to tell me how to do this step by step . thank you very much .

**Comment by bo0tzz** [comment]
Here's a guide you can use as a starting point. Of course you can use any service you like. https://immich.app/docs/guides/custom-map-styles

**Reply by Alberteins** [reply]
thank you very much for your reply .but , I am not trying to chang my map style. I want to change the map service provider ,because the https://[cloud.maptiler.com](https://cloud.maptiler.com/)/  can not be reached because of the internet issue.

**Reply by zackpollard** [reply]
The style also specifies where the tiles are downloaded from. However the default maps are served from https://tiles.mmich.cloudb not from map tiler. This uses cloudflare as our hosting provider

**Comment by zackpollard** [comment]
Out of interest, where are you located? We are always trying to improve the map service as we run it ourselves.

**Reply by at-wr** [reply]
In Mainland China, I guess (since the `【` `】`  bracket in the title is wildly used there).
By the way, China has strict restrictions on map providers, forcing the map providers to have their censorship passed by the government before that could be publicly served. Because of that policy, most apps that require map services use APIs from existing Chinese maps instead.
For instance, Apple has its own map services named Apple Maps, which previously used Google Maps as its map source, but later altered to its own source. In Mainland China, however, Apple uses Amap (a.k.a. AutoNavi) for the map source.
Google used to provide services in China, but later quit from its market due to multiple factors, mostly censorship from the government. If you compare a place in China with Google Map and any popular Chinese Map services, you can find significant differences, which is because the information in China from Google Map has been outdated as it stopped providing its map services at maps.google.cn.

**Reply by at-wr** [reply]
As the service situation is very different in China, I think it’s better to start a new thread about this topic.

**Comment by kangaroo72** [comment]
Hi... I can confirm this issue. Is it possible to change e. g. to openstreetmap or similar?
Thanks

**Comment by asheroto** [comment]
The map won't detect certain smaller town names, and the map also won't accept an address. I've found myself finding the location on Google Maps by looking at the streets, then clicking on the approximate area.

Although a _public_ Google Maps API option may not be available for use, Google Maps could still be utilized within Immich if users were to paste their Google Maps Platform API key in `Map & GPS Settings`. The API key is free. It would just need to support that functionality, of course. 😊

**Reply by noliveleger** [reply]
Same thing here. 9/10 it does not find the (small) cities (Canada and France) I'm looking for. Search by address would be a very nice plus. 
Moreover, the cities it refers to when you pick a point on a map are wrong. It's hard to search by city because the dropdown is populated with cities I even don't know. 


@asheroto, something that works for me:
- Search for the location in Google Maps
- Copy the GPS location from the marker (with right-click)
- Set the focus in the latitude field
- Paste the GPS location

Not ideal but it does work and it gives more accurate positions in Immich.

**Comment by jakern** [comment]
Could someone  please point me to where the tile server is set so I can have a look at making it configurable?

**Reply by bo0tzz** [reply]
It already is configurable, the tile server is set in the map style json and you can set a custom one of those at https://my.immich.app/admin/system-settings?isOpen=location+map.

**Comment by afonsofrancof** [comment]
Was going crazy trying to find out why my map wasn't loading.... (Cloudflare is down)
Will look into ways to self-host this, since it makes no sense for my own services to be down because of cloudflare.

**Comment by Test-subj** [comment]
In my case, I'm also running Reitti, which runs local tile and reverse geocode servers. I'd love to be able to point them at localhost:port and not use any external services at all.

**Comment by cedricwalter** [comment]
Immich cannot find small villages like "Mürren" because its text-based location naming relies on a pre-filtered offline database, not live OpenStreetMap data.While Immich displays your photos on an OpenStreetMap visual tile layer, it performs the actual text translation (turning coordinates into words like "Mürren") entirely offline using a dataset called GeoNames. 

By default, Immich only packages the cities500.txt file from GeoNames. This file ignores any location with a population under 500 people or areas that do not have their population figures strictly specified.Because Mürren has a permanent population of only aroundv500 people, Immich's offline database skips it completely and falls back to matching your photo to the closest town with >500 residents (such as Lauterbrunnen or Interlaken).
